crates/ruff_python_parser/src/lexer.rs

```diff
@@ -508,6 +541,161 @@ impl<'source> Lexer<'source> {
         }
     }
 
+    /// Lex a f-string start token.
+    #[cfg(feature = "pep-701")]
+    fn lex_fstring_start(&mut self, quote: char, is_raw_string: bool) -> Tok {
+        #[cfg(debug_assertions)]
+        debug_assert_eq!(self.cursor.previous(), quote);
+
+        // If the next two characters are also the quote character, then we have a triple-quoted
+        // string; consume those two characters and ensure that we require a triple-quote to close
+        let quote_size = if self.cursor.first() == quote && self.cursor.second() == quote {
+            self.cursor.bump();
+            self.cursor.bump();
+            StringQuoteSize::Triple
+        } else {
+            StringQuoteSize::Single
+        };
```

Comment:
This is repetitive from the `lex_string` function, maybe we could define a `consume_open_quote` function:

```rust
fn consume_open_quote(&mut self, quote: char) -> (StringQuoteChar, StringQuoteSize) {
	// ...
}
```

Or, return bitflags?